scalameta/semantic/shared/src/main/scala/scala/meta/semantic/Message.scala

```diff
@@ -0,0 +1,33 @@
+package scala.meta
+package semantic
+
+import org.scalameta.adt._
+import org.scalameta.data._
+import org.scalameta.unreachable
+
+@data class Message(anchor: Anchor, severity: Severity, message: String) {
```

Comment:
The fact that it's tied to an `Input` is exactly why I think it's a good idea. For example, `Anchor` does not include the file encoding like `Input.File`, which could be problematic in some cases.

`Input.LabeledString` would be a perfect fit for a `metadoc` tool I want to write, which generates a static site from a Mirror and uses Scala.js for interactive code browsing and symbol search.